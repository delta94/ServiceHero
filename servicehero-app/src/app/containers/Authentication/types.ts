import { JwtPayload } from 'types';

export interface AuthenticationState {
  accessToken: string | null;
  user: null | JwtPayload;
  loading: boolean;
  error?: null | string;
}

export interface LoginPayload {
  email: string;
  password: string;
}

export type ContainerState = AuthenticationState;
