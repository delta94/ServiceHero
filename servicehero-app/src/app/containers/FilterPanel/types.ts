import { FilterTypes } from 'types';

/* --- STATE --- */
export interface FilterPanelState {
  queryParam: string;
  filter: FilterTypes;
}

export type ContainerState = FilterPanelState;
