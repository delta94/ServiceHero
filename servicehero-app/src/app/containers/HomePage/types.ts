import { Listing } from 'types';

/* --- STATE --- */
export interface HomePageState {
  listings: Listing[];
  loading: boolean;
  error?: string | null;
}

export type ContainerState = HomePageState;
