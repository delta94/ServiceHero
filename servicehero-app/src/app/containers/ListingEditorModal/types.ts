import { Listing } from 'types';

/* --- STATE --- */
export interface ListingEditorModalState {
  isOpen: boolean;
  listing?: Listing | null;
}

export type ContainerState = ListingEditorModalState;
