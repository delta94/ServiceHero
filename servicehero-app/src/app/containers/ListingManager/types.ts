import { ListingDto } from 'types';
export interface ListingManagerState {
  isSuccess: boolean;
  loading: boolean;
  error?: null | string;
}

export interface EditListingPayload {
  id: string;
  listingDto: ListingDto;
}

export type ContainerState = ListingManagerState;
