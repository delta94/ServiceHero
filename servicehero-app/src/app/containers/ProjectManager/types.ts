import { ProjectStatus, Review } from 'types';

/* --- STATE --- */
export interface ProjectManagerState {
  isSuccess: boolean;
  loading: boolean;
  error?: null | string;
}

export interface UpdateStatusPayload {
  projectId: string;
  newStatus: ProjectStatus;
}

export interface UpdateReviewPayload {
  projectId: string;
  review: Review;
}

export type ContainerState = ProjectManagerState;
