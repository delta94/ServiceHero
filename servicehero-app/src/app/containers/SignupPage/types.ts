import { UserType } from 'types';

export interface UserDto {
  name: string;
  email: string;
  password: string;
  type: UserType;
}

export interface SignupPageState {
  isSuccess: boolean;
  loading: boolean;
  error?: string | null;
}

export type ContainerState = SignupPageState;
