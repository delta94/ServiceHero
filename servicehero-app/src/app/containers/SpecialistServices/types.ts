import { Listing } from 'types';

/* --- STATE --- */
export interface SpecialistServicesState {
  listings: Listing[];
  loading: boolean;
  error?: null | string;
}

export type ContainerState = SpecialistServicesState;
