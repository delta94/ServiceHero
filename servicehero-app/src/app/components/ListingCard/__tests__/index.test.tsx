import React from 'react';
import { render } from '@testing-library/react';
import { ListingCard } from '../index';
import { Listing, ListingType, UserType } from 'types';
import { Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import * as H from 'history';

import 'utils/mock/matchMedia';
import { configureAppStore } from 'store/configureStore';

const listing: Listing = {
  availability: ['18:00', '19:00'],
  _id: '600de6f7b8639085d641428d',
  type: ListingType.Cleaning,
  description: 'I am really good mover and do it cheaply',
  hourlyRate: 20,
  specialist: {
    _id: '600c7e3efd9e52514647b420',
    name: 'Martin',
    email: 'martin@email.com',
    type: UserType.Specialist,
  },
};

describe('<ListingCard  />', () => {
  it('should match snapshot', () => {
    const loadingIndicator = render(
      <Provider store={configureAppStore()}>
        <Router history={H.createBrowserHistory()}>
          <ListingCard loading={false} listing={listing} />
        </Router>
      </Provider>,
    );
    expect(loadingIndicator.container.firstChild).toMatchSnapshot();
  });
});
