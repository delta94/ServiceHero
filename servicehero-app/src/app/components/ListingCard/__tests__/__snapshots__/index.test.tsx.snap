// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListingCard  /> should match snapshot 1`] = `
.c4 {
  display: inline-block;
}

.c3 {
  margin-left: 4px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c5 {
  font-size: 1.2em;
  font-weight: bold;
}

.c1 {
  max-width: 300px;
  min-width: 300px;
  cursor: default;
  border-radius: 8px;
}

.c0 {
  margin: 8px;
}

<div
  class="c0"
>
  <div
    class="ant-card ant-card-bordered ant-card-hoverable c1"
  >
    <div
      class="ant-card-body"
    >
      <div
        class="ant-card-meta"
      >
        <div
          class="ant-card-meta-avatar"
        >
          <span
            class="ant-avatar ant-avatar-circle ant-avatar-icon"
          >
            <span
              aria-label="home"
              class="anticon anticon-home"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="home"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
                />
              </svg>
            </span>
          </span>
        </div>
        <div
          class="ant-card-meta-detail"
        >
          <div
            class="ant-card-meta-title"
          >
            <span
              class="c2"
            >
              Cleaning
               service by
              <a
                class="c3"
                href="/specialist/600c7e3efd9e52514647b420"
              >
                Martin
              </a>
            </span>
          </div>
          <div
            class="ant-card-meta-description"
          >
            <ul
              class="ant-rate ant-rate-disabled c4"
              role="radiogroup"
              tabindex="-1"
            >
              <li
                class="ant-rate-star ant-rate-star-zero"
              >
                <div
                  aria-checked="false"
                  aria-posinset="1"
                  aria-setsize="5"
                  role="radio"
                  tabindex="-1"
                >
                  <div
                    class="ant-rate-star-first"
                  >
                    <span
                      aria-label="star"
                      class="anticon anticon-star"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="star"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                        />
                      </svg>
                    </span>
                  </div>
                  <div
                    class="ant-rate-star-second"
                  >
                    <span
                      aria-label="star"
                      class="anticon anticon-star"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="star"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </li>
              <li
                class="ant-rate-star ant-rate-star-zero"
              >
                <div
                  aria-checked="false"
                  aria-posinset="2"
                  aria-setsize="5"
                  role="radio"
                  tabindex="-1"
                >
                  <div
                    class="ant-rate-star-first"
                  >
                    <span
                      aria-label="star"
                      class="anticon anticon-star"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="star"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                        />
                      </svg>
                    </span>
                  </div>
                  <div
                    class="ant-rate-star-second"
                  >
                    <span
                      aria-label="star"
                      class="anticon anticon-star"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="star"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </li>
              <li
                class="ant-rate-star ant-rate-star-zero"
              >
                <div
                  aria-checked="false"
                  aria-posinset="3"
                  aria-setsize="5"
                  role="radio"
                  tabindex="-1"
                >
                  <div
                    class="ant-rate-star-first"
                  >
                    <span
                      aria-label="star"
                      class="anticon anticon-star"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="star"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                        />
                      </svg>
                    </span>
                  </div>
                  <div
                    class="ant-rate-star-second"
                  >
                    <span
                      aria-label="star"
                      class="anticon anticon-star"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="star"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </li>
              <li
                class="ant-rate-star ant-rate-star-zero"
              >
                <div
                  aria-checked="false"
                  aria-posinset="4"
                  aria-setsize="5"
                  role="radio"
                  tabindex="-1"
                >
                  <div
                    class="ant-rate-star-first"
                  >
                    <span
                      aria-label="star"
                      class="anticon anticon-star"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="star"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                        />
                      </svg>
                    </span>
                  </div>
                  <div
                    class="ant-rate-star-second"
                  >
                    <span
                      aria-label="star"
                      class="anticon anticon-star"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="star"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </li>
              <li
                class="ant-rate-star ant-rate-star-zero"
              >
                <div
                  aria-checked="false"
                  aria-posinset="5"
                  aria-setsize="5"
                  role="radio"
                  tabindex="-1"
                >
                  <div
                    class="ant-rate-star-first"
                  >
                    <span
                      aria-label="star"
                      class="anticon anticon-star"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="star"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                        />
                      </svg>
                    </span>
                  </div>
                  <div
                    class="ant-rate-star-second"
                  >
                    <span
                      aria-label="star"
                      class="anticon anticon-star"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="star"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
              </li>
            </ul>
            <br />
            <span
              class="c5"
            >
              € 
              20
               / hour
            </span>
            <div
              class=""
            >
              I am really good mover and do it cheaply
            </div>
            <div
              class=""
            >
              Hours: 
              18:00, 19:00
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
